.claude/skills/ad-layer-visualizer/scripts/extract_trace_kernels.py

```
@@ -0,0 +1,448 @@
+
+
+def ensure_sqlite(trace_path: str) -> str:
+    """Convert .nsys-rep to .sqlite if needed, return sqlite path."""
+    p = Path(trace_path)
+    if p.suffix == ".sqlite":
+        return trace_path
+    sqlite_path = p.with_suffix(".sqlite")
+    if sqlite_path.exists():
+        return str(sqlite_path)
+    print(f"Exporting {p.name} to sqlite...", file=sys.stderr)
+    subprocess.run(
```

Ruff (0.15.13)

[error] `subprocess` call: check for execution of untrusted input

(S603)